Add support for multiple rest sensors with a single endpoint (#16632)
home-assistant/home-assistant.github.io: source/_integrations/rest.markdown

---
title: RESTful

> description: Instructions on how to setup rest sensors within Home Assistant.

Reviewer: ```suggestion
description: Instructions on how to set up rest sensors within Home Assistant.
```

ha_category:
  - Sensor
  - Binary Sensor

> ha_release: 2021.3

Reviewer: This doesn't change :)
```suggestion
ha_release: 0.7.4
```

The `rest` sensor platform is consuming a given endpoint which is exposed by a [RESTful API](https://en.wikipedia.org/wiki/Representational_state_transfer) of a device, an application, or a web service. The sensor has support for GET and POST requests.

> [RESTful Sensor](/integrations/sensor.rest) and [RESTful Binary_sensor](/integrations/binary_sensor.rest) can also be setup as platforms if there is only a single sensor per endpoint.

Reviewer: ```suggestion
[RESTful Sensor](/integrations/sensor.rest) and [RESTful Binary_sensor](/integrations/binary_sensor.rest) can also be set up as platforms if there is only a single sensor per endpoint.
```

```yaml
# Example configuration.yaml entry
rest:

>   - authentication: basic
>     username: admin
>     password: password
>     scan_interval: 60
>     resource: http://192.168.1.12/status.xml

Reviewer: ```suggestion
  - authentication: basic
    username: "admin"
    password: "password"
    scan_interval: 60
    resource: http://192.168.1.12/status.xml
```

    sensor:

>       - name: Adult Pool Data System

Reviewer: ```suggestion
      - name: "Adult Pool Data System"
```

        json_attributes_path: "$.response.system"

>         value_template: 'OK'

Reviewer: ```suggestion
        value_template: "OK"
```

        json_attributes:
          - "runstate"
          - "model"
          - "opmode"
          - "freeze"
          - "time"
          - "sensor1"
          - "sensor2"
          - "sensor3"
          - "sensor4"
          - "sensor5"
          - "version"

>       - name: Adult Pool Data Equipment
>         json_attributes_path: "$.response.equipment"
>         value_template: 'OK'

Reviewer: ```suggestion
      - name: "Adult Pool Data Equipment"
        json_attributes_path: "$.response.equipment"
        value_template: "OK"
```

        json_attributes:
          - "circuit1"
          - "circuit2"
          - "circuit3"
          - "circuit4"
          - "circuit5"
          - "circuit6"
          - "circuit7"
          - "circuit8"

>       - name: Adult Pool Data Temp
>         json_attributes_path: "$.response.temp"
>         value_template: 'OK'

Reviewer: ```suggestion
      - name: "Adult Pool Data Temp"
        json_attributes_path: "$.response.temp"
        value_template: "OK"
```

        json_attributes:
          - "htstatus"
          - "poolsp"
          - "spasp"
          - "pooltemp"
          - "spatemp"
          - "airtemp"
  - authentication: basic

>     username: admin
>     password: password
>     scan_interval: 60
>     resource: http://192.168.1.13/status.xml

Reviewer: ```suggestion
    username: "admin"
    password: "password"
    scan_interval: 60
    resource: "http://192.168.1.13/status.xml"
```

    sensor:

>       - name: Kiddie Pool Data System
>         json_attributes_path: "$.response.system"
>         value_template: 'OK'

Reviewer: ```suggestion
      - name: "Kiddie Pool Data System"
        json_attributes_path: "$.response.system"
        value_template: "OK"
```

        json_attributes:
          - "runstate"
          - "model"
          - "opmode"
          - "freeze"
          - "time"
          - "sensor1"
          - "sensor2"
          - "sensor3"
          - "sensor4"
          - "version"

>       - name: Kiddie Pool Data Equipment
>         json_attributes_path: "$.response.equipment"
>         value_template: 'OK'

Reviewer: ```suggestion
      - name: "Kiddie Pool Data Equipment"
        json_attributes_path: "$.response.equipment"
        value_template: "OK"
```

        json_attributes:
          - "circuit1"
          - "circuit2"
          - "circuit3"
          - "circuit4"
          - "circuit5"
          - "circuit6"
          - "circuit7"
          - "circuit8"

>       - name: Kiddie Pool Data Temp
>         json_attributes_path: "$.response.temp"
>         value_template: 'OK'

Reviewer: ```suggestion
      - name: "Kiddie Pool Data Temp"
        json_attributes_path: "$.response.temp"
        value_template: "OK"
```

params:
  description: The query params for the requests.
  required: false
  type: [string, list]
sensor:

>   description: A list of [RESTful Sensor](/integrations/sensor.rest) to create from the shared data. All configuration settings that the supported by `RESTful Sensor` not listed above can be used here.

Author: ```suggestion
  description: A list of [RESTful Sensor](/integrations/sensor.rest) to create from the shared data. All configuration settings that the supported by [RESTful Sensor](/integrations/sensor.rest#configuration-variables) not listed above can be used here.
```

  required: false
  type: list
binary_sensor:

>   description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.

Author: ```suggestion
  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest#configuration-variables) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.
```

>   description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.

Author: ```suggestion
  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by [RESTful Binary_sensor](/integrations/binary_sensor.rest#configuration-variables) not listed above can be used here.
```
